code/game/machinery/frame.dm

```diff
@@ -0,0 +1,307 @@
+						src.state = 2
+						src.icon_state = "[frame_type]_2"
+						var/obj/item/stack/cable_coil/A = new /obj/item/stack/cable_coil( src.loc )
+						A.amount = 5
+					if(istype(P, /obj/item/stack/material) && P.get_material_name() == "glass")
+						var/obj/item/stack/G = P
+						if (G.get_amount() < 2)
+							user << "<span class='warning'>You need two sheets of glass to put in the glass panel.</span>"
+							return
+						playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
+						user << "<span class='notice'>You start to put in the glass panel.</span>"
+						if(do_after(user, 20) && state == 3)
+							if (G.use(2))
+								user << "<span class='notice'>You put in the glass panel.</span>"
+								src.state = 4
+								src.icon_state = "[frame_type]_4"
+
+				if("display")  //computer and display steps
+					if(istype(P, /obj/item/weapon/wirecutters))
+						playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+						user << "<span class='notice'>You remove the cables.</span>"
+						src.state = 2
+						src.icon_state = "[frame_type]_2"
+						var/obj/item/stack/cable_coil/A = new /obj/item/stack/cable_coil( src.loc )
+						A.amount = 5
+					if(istype(P, /obj/item/stack/material) && P.get_material_name() == "glass")
+						var/obj/item/stack/G = P
+						if (G.get_amount() < 2)
+							user << "<span class='warning'>You need two sheets of glass to put in the glass panel.</span>"
+							return
+						playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
+						user << "<span class='notice'>You start to put in the glass panel.</span>"
+						if(do_after(user, 20) && state == 3)
+							if (G.use(2))
+								user << "<span class='notice'>You put in the glass panel.</span>"
+								src.state = 4
+								src.icon_state = "[frame_type]_4"
+
+				if("machine") //machine steps
+					if(istype(P, /obj/item/weapon/wirecutters))
+						playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+						user << "<span class='notice'>You remove the cables.</span>"
+						src.state = 2
+						src.icon_state = "[frame_type]_2"
+						var/obj/item/stack/cable_coil/A = new /obj/item/stack/cable_coil( src.loc )
+						A.amount = 5
+					if(istype(P, /obj/item/weapon/crowbar))
+						playsound(src.loc, 'sound/items/Crowbar.ogg', 50, 1)
+						if(components.len == 0)
+							user << "<span class='notice'>There are no components to remove.</span>"
+						else
+							user << "<span class='notice'>You remove the components.</span>"
+							for(var/obj/item/weapon/W in components)
+								W.loc = src.loc
+						components = list()
+						req_components = circuit.req_components.Copy()
+						for(var/A in circuit.req_components)
+							req_components[A] = circuit.req_components[A]
+						req_component_names = circuit.req_components.Copy()
+						for(var/A in req_components)
+							var/cp = text2path(A)
+							var/obj/ct = new cp() // have to quickly instantiate it get name
+							req_component_names[A] = ct.name
```

Comment:
text2path() shouldn't be needed and creating a whole new object just to get the name is hopefully excessive. Instead this may do
```
var/obj/ct = A
req_component_names[A] = initial(ct.name)
```
Won't work if ct changes its name in New()/initialize() but should otherwise work wonders without having to create the object.